static/app/components/links/listLink.tsx

```diff
@@ -24,23 +25,28 @@ type Props = DefaultProps &
 class ListLink extends React.Component<Props> {
   static displayName = 'ListLink';
 
-  static contextTypes = {
-    router: PropTypes.object.isRequired,
-  };
-
   static defaultProps: DefaultProps = {
     activeClassName: 'active',
     index: false,
     disabled: false,
   };
 
-  isActive = () => {
-    const {isActive, to, query, index} = this.props;
+  isActive() {
+    const {isActive, to, query, index, location, router} = this.props;
+    const queryData = query ? qs.parse(query) : undefined;
+    const target: LocationDescriptor =
```

Comment:
Instead of doing this logic, it looks like it was relying solely on `to` being `Path` (aka. a string) instead of a LocationDescriptorObject or a function. I guess alternatively, can we just modify the props of this ListLink to only accept `to` as a Path and leverage ts to make sure it's used correctly everywhere?

Con: it has less utility. But the pro is that, well, if we end up doing a bunch of type discrimination and handling as a pattern in general, it's a lot of extra code that might not even be used, so it's more lightweight to only handle the part of the union you actually need.